### PR TITLE
Sidekick/qmk firmware

### PR DESCRIPTION
**Pull request by AlisGraveNil**
## Description

It's a 27-key macropad designed to go with a 60% or smaller keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by AlisGraveNil** [comment]
I only wanted to add a new macropad to the keyboard list; I have no idea how I managed to mess up the "Travis CI" build thing

File: keyboards/clawsome/sidekick/rules.mk [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (tzarc) [inline keyboards/clawsome/sidekick/rules.mk:11]:
```suggestion
```

File: keyboards/clawsome/sidekick/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
```
Comment (tzarc) [inline keyboards/clawsome/sidekick/config.h:43]:
```suggestion
```

**Comment by tzarc** [comment]
As discussed on Discord, please update the PR with a description, as well as ticking the appropriate boxes in the form.

File: keyboards/clawsome/sidekick/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/clawsome/sidekick/rules.mk:4]:
```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/clawsome/sidekick/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no             # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no         # Enable keyboard RGB underglow
+MIDI_ENABLE = no             # MIDI support
+UNICODE_ENABLE = no          # Unicode
+BLUETOOTH_ENABLE = no        # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no            # Audio output on port C6
+FAUXCLICKY_ENABLE = no       # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no          # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/clawsome/sidekick/rules.mk:26]:
```suggestion
```

File: keyboards/clawsome/sidekick/info.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{
+    "keyboard_name": "Sidekick",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 7,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 27,
```
Comment (fauxpark) [inline keyboards/clawsome/sidekick/info.json:9]:
```suggestion
```

File: keyboards/clawsome/sidekick/info.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                {"label":"K36 (B6,D4)", "x":6, "y":3, "h":2},
+                {"label":"K40 (B5,F6)", "x":0, "y":4},
+                {"label":"K41 (B5,F5)", "x":1, "y":4},
+                {"label":"K42 (B5,B1)", "x":2, "y":4},
+                {"label":"K43 (B5,B3)", "x":3, "y":4, "w":2},
+                {"label":"K45 (B5,D7)", "x":5, "y":4}
+            ]
+        }
+    }
+    ,"meta": "https://noroadsleft.github.io/kbf_qmk_converter/"
```
Comment (fauxpark) [inline keyboards/clawsome/sidekick/info.json:41]:
```suggestion
```

File: keyboards/clawsome/sidekick/config.h [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0000
```
Comment (fauxpark) [inline keyboards/clawsome/sidekick/config.h:24]:
```suggestion
#define PRODUCT_ID      0xDB9F
```
```
echo -n "Sidekick" | shasum | cut -c1-4
```

File: keyboards/clawsome/sidekick/rules.mk (hunk 1 above)
Comment (drashna) [reply keyboards/clawsome/sidekick/rules.mk:11]: Since it is advertised as supporting a ProMicro/Elite C, I would rather the list be left here.

**Pull request merged**